src/aiida/tools/dumping/processes.py

```diff
@@ -285,6 +289,53 @@ def _dump_calculation(
                 link_triples=output_links,
             )
 
+        # Additional remote file retrieval should only apply for CalcJobNodes, not CalcFunctionNodes
+        if self.also_remote and isinstance(calculation_node, CalcJobNode):
+            self._dump_calculation_remote_files(calcjob_node=calculation_node, output_path=output_path)
+
+    def _dump_calculation_remote_files(self, calcjob_node: CalcJobNode, output_path: Path) -> None:
+        """Dump the additional remote files attached to a `CalcJobNode` to a specified output path.
+
+        :param calcjob_node: The `CalcJobNode` to be dumped.
+        :param output_path: The path where the files will be dumped.
+        """
+
+        remote_workdir = calcjob_node.get_remote_workdir()
+        if remote_workdir is None:
+            raise NotExistentAttributeError(f"CalcJobNode <{calcjob_node.pk}> doesn't have a `remote_workdir`.")
+
+        # Exclude the objects that were already dumped, as they were either originally retrieved via `retrieve_list`
+        # or are already part of the file repository of the CalcJobNode, e.g. the `aiida.in` and `_aiidasubmit.sh`
+        retrieve_list = list(calcjob_node.get_retrieve_list())  # type: ignore[arg-type]
+        repository_list = calcjob_node.base.repository.list_object_names()
+        exclude_list = retrieve_list + repository_list
```

Comment:
Few things here:

1. Shouldn't everything that it is in the retrieve list _also_ be in the repository list? Maybe there may be more in the retrieve list, but that would then correspond to a file that doesn't exist, so it will presumable still not be there on the remote, so it can be ignored.
2. a bigger problem though I think is that the `retrieve_list` can contain wildcards and I don't think you currently handle them in the code below right? If you just rely on the contents of the repository, this problem disappears.